### ISSUE TITLE
Spotify Connect on Android not detecting Sonos speakers

### ISSUE THREAD

**Issue by eleete0712** [StevenBlack/hosts#619]
Hello everyone,

some time ago the Spotify app on Android stopped showing up my Sonos speakers when looking up Connect enabled speakers. I tried to pinpoint the issue down to either Spotify or Sonos and also talked to both of their support teams which didn't yield any results. I figured that when using an older version of the Spotify app (8.4.24.871) the Sonos speakers would show up again. I told them their app was messed up and that there must have been some change in the implementation. They couldn't reproduce the issue and I was on my own again.
Yesterday I noticed that when I disabled Adblock (Magisk - Unified Hosts Adblock - uses this list) the Sonos speakers would start showing up again inside the Spotify app. So I tried to pinpoint it down to some Spotify or Sonos rules in this list but there is only a single one which isn't the cause for the issue. Next I tried commenting several lines from the top and I found the line that seems to cause the issue:

`::1 localhost`

By default my Android device only has the following two lines in the hosts file:

`127.0.0.1 localhost`
`::1 ip6-localhost`

So it seems that this line is kind of problematic on Android or at least for the Android, Spotify and Sonos combo. I don't know if this causes other issues or causes issues if left out but for me it makes things work perfectly fine again.

Regards

**Comment by StevenBlack** [comment]
Hi @eleete0712 I jumped on `msmetrics.ws.sonos.com` a few weeks ago when I noticed it being accessed repeatedly (dozens of times per hour) from my own home network.

Everything functions normally here, on iOS and Mac...

It seems odd that `::1 localhost` would be problematic.  Anyone have any insight on this?

**Comment by eleete0712** [comment]
Hi @StevenBlack,
I noticed the `msmetrics.ws.sonos.com` line which I also removed before the `::1 localhost` line without success.
I used Wireshark to find out what was going on. I noticed that my Laptop (Windows and Linux) with Spotify sent out way more mDNS queries and some that my Android device did not send at all (mDNS to `224.0.0.251` with content `_spotify-connect._tcp.local'). After commenting `::1 localhost` those mDNS packages are sent again.
But I'm not too much into networking that I could explain that.
